### ISSUE TITLE
import/order error not detected

### ISSUE THREAD

**Issue by caralin3** [import-js/eslint-plugin-import#3178]
I have a component library that uses react and materiaul-ui. When I try to lint my project, the import/order rule error is not detected, unless I add `import React from 'react'` in my component file (which is unused). That import is not really needed. How can I get eslint to detect the import order error without that import?

Actual errors (none) without react import
```
Done in 1.38s.
```

Expected errors without react import
```
/Users/cara/Documents/Personal Projects/notes-app/packages/ui-library/src/components/Button/Button.tsx
  3:1  error    There should be at least one empty line between import groups        import/order
  6:1  error    `react` import should occur before import of `@mui/material/Button`  import/order
  6:8  warning  'React' is defined but never used                                    @typescript-eslint/no-unused-vars

```

```
// project structure

ui-library
|
+-- src
|   |
|   +-- components
|   |   |
|   |   +-- Button
|   |   |   |
|   |   |   +-- Button.tsx
+-- eslint.config.mjs
+-- package.json
+-- tsconfig.json
```

```
// eslint.config.mjs

import js from '@eslint/js';
import globals from 'globals';
import eslintConfigPrettier from 'eslint-config-prettier';
import importPlugin from 'eslint-plugin-import';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import pluginReact from 'eslint-plugin-react';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  { ignores: ['dist'] },
  {
    files: ['**/*.{ts,tsx}'],
    plugins: { js, 'react-hooks': reactHooks, 'react-refresh': reactRefresh },
    languageOptions: { ecmaVersion: 2020, globals: globals.browser },
    extends: [
      js.configs.recommended,
      tseslint.configs.recommended,
      pluginReact.configs.flat.recommended,
      importPlugin.flatConfigs.recommended,
      eslintConfigPrettier,
    ],
    rules: {
      ...reactHooks.configs.recommended.rules,
      'react-refresh/only-export-components': [
        'warn',
        { allowConstantExport: true },
      ],
      '@typescript-eslint/no-unused-vars': 'warn',
      'import/no-unresolved': ['error', { ignore: ['\\.svg\\?url$'] }],
      // this is for sorting WITHIN an import
      'sort-imports': [
        'error',
        { ignoreCase: true, ignoreDeclarationSort: true },
      ],
      // this is for sorting imports
      'import/order': [
        'error',
        {
          groups: [
            ['external', 'builtin'],
            'internal',
            ['sibling', 'parent'],
            'index',
          ],
          pathGroups: [
            {
              pattern: '@(react|react-native)',
              group: 'external',
              position: 'before',
            },
            {
              pattern: '^[a-zA-Z]',
              group: 'internal',
            },
            {
              pattern: '^@/?\\w',
              group: 'internal',
            },
            {
              pattern: '^[./]',
              group: 'internal',
            },
          ],
          pathGroupsExcludedImportTypes: ['internal', 'react'],
          'newlines-between': 'always',
          alphabetize: {
            order: 'asc',
            caseInsensitive: true,
          },
        },
      ],
    },
    settings: {
      react: {
        version: 'detect',
      },
      'import/resolver': {
        typescript: {
          project: './tsconfig.json',
        },
      },
    },
  },
);
```

```
// package.json

{
  "name": "ui-library",
  "private": true,
  "version": "1.0.0",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "watch": "rollup -c --watch --bundleConfigAsCjs",
    "build": "rollup -c --bundleConfigAsCjs"
  },
  "devDependencies": {
    "@emotion/react": "^11.14.0",
    "@emotion/styled": "^11.14.0",
    "@eslint/js": "^9.24.0",
    "@mui/icons-material": "^7.0.2",
    "@mui/material": "^7.0.2",
    "@rollup/plugin-commonjs": "^28.0.3",
    "@rollup/plugin-node-resolve": "^16.0.1",
    "@rollup/plugin-typescript": "^12.1.2",
    "@types/react": "^19.0.9",
    "eslint": "^9.24.0",
    "eslint-config-prettier": "^10.1.2",
    "eslint-import-resolver-typescript": "^4.3.2",
    "eslint-plugin-import": "^2.31.0",
    "eslint-plugin-react": "^7.37.5",
    "eslint-plugin-react-hooks": "^5.2.0",
    "eslint-plugin-react-refresh": "^0.4.19",
    "globals": "^16.0.0",
    "prettier": "^3.5.3",
    "react": "^19.0.0",
    "rollup": "^4.39.0",
    "rollup-plugin-dts": "6.1.0",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "rollup-plugin-terser": "^7.0.2",
    "typescript": "^5.8.3",
    "typescript-eslint": "^8.29.1"
  },
  "peerDependencies": {
    "react": "^19.0.0"
  }
}
```

```
// tsconfig.json

{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "jsx": "react-jsx",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "strict": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "sourceMap": true,
    "noEmit": true
  }
}

```

**Comment by caralin3** [comment]
Nevermind, after I committed the changes, it worked without the react import.
